File: Python/Practice/LeetCode/Easy/Count Asterisks.py
# https://leetcode.com/problems/count-asterisks/description/

class Solution:
    def countAsterisks(self, s: str) -> int:
        isInsideBars = False
        counter = 0
        for i in range(len(s)):
            if (s[i] == "|"):
                isInsideBars = not isInsideBars
            if not isInsideBars and s[i] == "*":
                counter += 1
        return counter

File: Java/Practice/LeetCode/Easy/Count Asterisks.java
// https://leetcode.com/problems/count-asterisks/description/

class Solution {
    public int countAsterisks(String s) {
        boolean isInsideBars = false;
        int counter = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '|') {
                isInsideBars = !isInsideBars;
            }
            if (!isInsideBars && s.charAt(i) == '*') {
                counter++;
            }
        }
        return counter;   
    }
}